bip-p2qrh.mediawiki

```diff
@@ -0,0 +1,240 @@
+
+Signature verification speed as it compares to Schnorr or ECDSA isn't seen as high a consideration as signature size due to block space being the primary fee constraint. As a P2QRH implementation materializes, a benchmark will be added for performance comparison. Fortunately, SQIsign signatures are substantially faster to verify than it is to generate keys or to sign, which is a major consideration when a transaction need only be signed once, or a handful of times with PSBT, compared to being verified simultaneously on tens of thousands of nodes. Key generation may need to cached in [https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki BIP-32 HD wallets].
+
+An additional consideration is security levels. Longer signature sizes provide more security. NIST has standardized five security levels for post-quantum cryptography. NIST security level I provides security equivalent to 128-bit keys, and security level V provides 256-bit security.
+
+
+== Specification ==
+
+How the quitness is differentiated from the witness can be accomplished similar to how [https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#user-content-Transaction_ID BIP-141] introduced the marker and flag, with the QuBit flag being set to 0x02. This means all QuBit transactions are also SegWit transactions. The additional data would be included as a second array of byte arrays following the witness stack.
+
+The new transaction serialization format would be as follows:
+
+  [nVersion][marker][flag][txins][txouts][witness][quitness][nLockTime]
+
+WIP
+
+=== Public Key Generation ===
+
+TBD, pending test vectors
+
+=== Public Key Conversion ===
+
+TBD
+
+=== Default Signing ===
+
+TBD
+
+=== Alternative Signing ===
+
+TBD
+
+=== Verification ===
+
+TBD
+
+=== Batch Verification ===
+
+TBD
+
+=== Usage Considerations ===
+
+TBD
+
+== Test Vectors and Reference Code ==
+
+TBD
+
+
+== References ==
+
+* [https://groups.google.com/g/bitcoindev/c/Aee8xKuIC2s/m/cu6xej1mBQAJ Mailing list discussion]
+* [https://delvingbitcoin.org/t/proposing-a-p2qrh-bip-towards-a-quantum-resistant-soft-fork/956?u=cryptoquick Delving Bitcoin discussion]
+* [https://bitcoinops.org/en/newsletters/2024/06/14/ Bitcoin OpTech newsletter]
+* [https://bitcoinops.org/en/podcast/2024/06/18/#draft-bip-for-quantum-safe-address-format Bitcoin OpTech discussion transcript]
+
+<references />
+
+
+== Changelog ==
+
+To help implementors understand updates to this BIP, we keep a list of substantial changes.
+
+* 2024-06: High level rough draft
+* 2024-07: Additional algorithms in PQC table
+* 2024-08: Add FALCON signatures, update to use NIST standard terminology, add public key sizes.
+* 2024-09: Additional detail on P2QS. Deprecate P2QR. Postpone SQIsign. Add details on quitness.
```

Comment:
A few non-urgent nits regarding the changelog:

- latest version first
- use [semvar](https://semver.org/); for the final draft, maybe these should be collapsed to one initial version entry
- YYYY-MM-DD date format

(See https://keepachangelog.com and https://www.gnu.org/prep/standards/html_node/Style-of-Change-Logs.html#Style-of-Change-Logs)

Reply:
Sounds good. Let me know if my latest changes have resolved this.

Reply:
Looks good for now, thank you.